taca/element/Element_Runs.py

```diff
@@ -446,17 +446,301 @@ def rsync_successful(self):
+                        continue
+                    # Filter by lane
+                    unassigned_indexes = [unassigned_index for unassigned_index in unassigned_indexes if unassigned_index["Lane"] == lane]
+                    # Remove overlapped indexes from the list of max_unassigned_indexes
+                    idx1_overlapped_len = min([demux_lens_pair[1] for demux_lens_pair in sub_demux_index_lens if demux_lens_pair[0] == sub_demux][0][0],
+                                              [demux_lens_pair[1] for demux_lens_pair in sub_demux_index_lens if demux_lens_pair[0] == sub_demux_with_max_index_lens][0][0])
+                    idx2_overlapped_len = min([demux_lens_pair[1] for demux_lens_pair in sub_demux_index_lens if demux_lens_pair[0] == sub_demux][0][1],
+                                              [demux_lens_pair[1] for demux_lens_pair in sub_demux_index_lens if demux_lens_pair[0] == sub_demux_with_max_index_lens][0][1])
+                    for unassigned_index in unassigned_indexes:
+                        idx1_overlapped_seq = unassigned_index['I1'][:idx1_overlapped_len]
+                        idx2_overlapped_seq = unassigned_index['I2'][:idx2_overlapped_len]
+                        # Remove the overlapped record from the max_unassigned_indexes list
+                        max_unassigned_indexes = [max_unassigned_index for max_unassigned_index in max_unassigned_indexes if not (max_unassigned_index['I1'][:idx1_overlapped_len] == idx1_overlapped_seq and max_unassigned_index['I2'][:idx2_overlapped_len] == idx2_overlapped_seq)]
+            # Append to the aggregated_unassigned_indexes list
+            aggregated_unassigned_indexes += max_unassigned_indexes
+        # Sort aggregated_unassigned_indexes list first by lane and then by Count in the decreasing order
+        aggregated_unassigned_indexes = sorted(aggregated_unassigned_indexes, key=lambda x: (x['Lane'], -int(x['Count'])))
+        # Write to a new UnassignedSequences.csv file under demux_dir
+        aggregated_unassigned_csv = os.path.join(self.run_dir, self.demux_dir, "UnassignedSequences.csv")
+        write_to_csv(aggregated_unassigned_indexes, aggregated_unassigned_csv)
+
+
+    # Aggregate demux results
+    def aggregate_demux_results(self, demux_results_dirs):
+        # Ensure the destination directory exists
+        if not os.path.exists(os.path.join(self.run_dir, self.demux_dir)):
+            os.makedirs(os.path.join(self.run_dir, self.demux_dir))
+        # Clear all content under dest_dir
+        clear_dir(os.path.join(self.run_dir, self.demux_dir))
+        demux_runmanifest = collect_demux_runmanifest(demux_results_dirs)
+        # Aggregate the output FastQ files of samples from multiple demux
+        aggregate_sample_fastq(demux_runmanifest)
+        # Symplink the output FastQ files of undet only if a lane does not have multiple demux
```

Comment:
So what do we do when the undetermined have multiple demux? Maybe we don't care about the undet fastq then?

Reply:
We ignore the FastQ files of undet from the lanes with multiple demux involved. But the statistics will be fixed by the function `aggregate_stats_unassigned`

Reply:
Yeah I think that makes sense since the undetermined can be overlapping with samples form other demultiplexing.